### PR TITLE
[castai-agent] add DNS Policy and hostnetwork in values

### PR DESCRIPTION
**Pull request by CastAIPhil**

### CODE REVIEW

File: charts/castai-agent/Chart.yaml [hunk 1]
```diff
@@ -2,5 +2,5 @@ apiVersion: v2
 name: castai-agent
 description: CAST AI agent deployment chart.
 type: application
-version: 0.58.5
+version: 0.58.6
```
Comment (fals) [inline charts/castai-agent/Chart.yaml:5]: We use SemVersion, minor is only for bug fixes

File: charts/castai-agent/templates/deployment.yaml [hunk 2]
```diff
@@ -142,17 +148,6 @@ spec:
...
-            - name: {{ $k }}
-              value: "{{ $v }}"
-          {{- end }}
-          envFrom:
-            - secretRef:
-          {{- if .Values.apiKey }}
-                name: {{ include "castai-agent.fullname" . -}}
-          {{- else }}
-                name: {{ required "apiKey or apiKeySecretRef must be provided" .Values.apiKeySecretRef -}}
-          {{- end }}
```
Comment (jansyk13) [inline charts/castai-agent/templates/deployment.yaml:155]: Deleted by accident?

Comment (CastAIPhil) [reply charts/castai-agent/templates/deployment.yaml:155]: No, linter was complaining about duplicate code, this code exists at line: 81 also

Comment (CastAIPhil) [reply charts/castai-agent/templates/deployment.yaml:155]: You're right, it was for a different container within the pod 🤦  re-added this code.

**Review by jansyk13 — APPROVED** [review]

**Pull request merged**